packages/snaps-rpc-methods/src/permitted/scheduleBackgroundEvent.ts

```diff
@@ -58,12 +58,15 @@ const offsetRegex = /Z|([+-]\d{2}:?\d{2})$/u;
 const ScheduleBackgroundEventsParametersStruct = object({
   date: refine(string(), 'date', (val) => {
```

Comment:
Hmm, wondering if it would be simpler to support either `duration` or `date` as a key? That may simplify our implementation and be nicer for devs. @Mrtenz WDYT from an API perspective?

Reply:
The only thing is if I make the params a union, superstruct will no longer return a detailed error. We just get the top level union error string.

Reply:
We can use `selectiveUnion` to pick a struct based on the `duration` or `date` key to prevent that from being a problem!